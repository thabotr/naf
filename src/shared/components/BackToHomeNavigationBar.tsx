import React from 'react';
import {ViewProps} from 'react-native';
import {Surface} from 'react-native-paper';
import {useThemedStyles} from '../providers/theme';
import globalStyles, {globalThemedStyles} from '../styles';
import IconButton from './IconButton';

type Props = ViewProps & {
  onBackToHome?: () => void;
};

export default function (props: Props): JSX.Element {
  const themedStyle = useThemedStyles(globalThemedStyles);
  return (
    <Surface {...props} style={themedStyle.navbar}>
      <IconButton
        icon="arrow-left"
        accessibilityLabel="back to home"
        onPress={props.onBackToHome}
        style={globalStyles.square}
      />
    </Surface>
  );
}
