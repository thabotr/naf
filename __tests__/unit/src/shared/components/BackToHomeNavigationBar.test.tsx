import {fireEvent, render, screen} from '@testing-library/react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import BackToHomeNavigationBar from '../../../../../src/shared/components/BackToHomeNavigationBar';
import themed from '../../../utils/themed';

describe('BackToHomeNavigationBar', () => {
  it('renders correctly', () => {
    const tree = renderer.create(themed(<BackToHomeNavigationBar />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("calls the 'onBackToHome' prop when the back to home button is clicked", () => {
    const mockOnBackToHome = jest.fn().mockName('mockOnBackToHome');
    render(themed(<BackToHomeNavigationBar onBackToHome={mockOnBackToHome} />));
    const backToHomeButton = screen.getByLabelText('back to home');
    fireEvent.press(backToHomeButton);
    expect(mockOnBackToHome).toHaveBeenCalledTimes(1);
  });
});
