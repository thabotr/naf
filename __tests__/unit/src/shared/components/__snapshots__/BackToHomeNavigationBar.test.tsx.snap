// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackToHomeNavigationBar renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#87cefa",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingRight": 15,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
    }
  }
>
  <View
    accessibilityLabel="back to home"
    accessibilityRole="button"
    accessibilityState={
      {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={false}
    hitSlop={
      {
        "bottom": 6,
        "left": 6,
        "right": 6,
        "top": 6,
      }
    }
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "overflow": "hidden",
        },
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
            "margin": 6,
            "overflow": "hidden",
          },
          {
            "borderRadius": 18,
            "height": 36,
            "width": 36,
          },
          undefined,
          {
            "borderRadius": 0,
          },
        ],
      ]
    }
  >
    <View>
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "#000000",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
            {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        Û∞Åç
      </Text>
    </View>
  </View>
</View>
`;
